tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -168,6 +169,12 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
                         if (localSpanContext.RawTraceId == spanContext.RawTraceId)
                         {
                             localSpanContext.AdditionalW3CTraceState += spanContext.AdditionalW3CTraceState;
+
+                            if (localSpanContext.RawSpanId == spanContext.RawSpanId && !string.IsNullOrEmpty(spanContext.LastParentId) && !string.Equals(spanContext.LastParentId, ZeroLastParentId))
```

Comment:
We are already checking if `localSpanContext.RawSpanId == spanContext.RawSpanId` above, so checking again is redundant.
```suggestion
                            if (!string.IsNullOrEmpty(spanContext.LastParentId) && !string.Equals(spanContext.LastParentId, ZeroLastParentId))
```

Reply:
So in the above we only checked the **RawTraceId**, but for the RFC Phase 3 we only add the parent if the **RawSpanId**  both match, WDYT? https://docs.google.com/document/d/1Zc7uAQTJ2G5vHEbmx8k4vMnYdQ-BnqZxewXk0ohgd3g/edit#heading=h.53cixy7xk9rp

Reply:
Oh, my mistake! I they were all `RawTraceId`. Didn't notice it was `RawSpanId` in the inner `if`. Ignore! 😅